### ISSUE TITLE
Current build may be broken?

### ISSUE THREAD

**Issue by nivintw** [markokajzer/discord-soundbot#87]
Hi there,
Trying to run the bot using the docker commands in the readme with a clean install results in:
```
docker.io/markokajzer/discord-soundbot:latest
yarn run v1.22.4
$ node dist/bin/soundbot.js
/app/node_modules/lowdb/adapters/FileSync.js:42
          throw e;
          ^

Error: EISDIR: illegal operation on a directory, read
    at Object.readSync (fs.js:524:3)
    at tryReadSync (fs.js:349:20)
    at readFileSync (fs.js:386:19)
    at FileSync.read (/app/node_modules/lowdb/adapters/FileSync.js:35:22)
    at LodashWrapper.db.read (/app/node_modules/lowdb/lib/main.js:32:21)
    at Object.module.exports [as default] (/app/node_modules/lowdb/lib/main.js:51:13)
    at Object.<anonymous> (/app/dist/src/util/db/connection.js:9:35)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32) {
  errno: -21,
  syscall: 'read',
  code: 'EISDIR'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
After I ran into the above I tried cloning the repo and poking around (running `docker-compose up`) and that also failed. The bot sounds really neat and I was looking forward to running it. Any ideas?

**Comment by markokajzer** [comment]
Definitely working on my machine I just tried it out again.

EISDIR means that something was performed on a directory, where a file was expected. Are you sure you did all of the necessary steps? I recommend create a new directory, e.g. `soundbot`, the creating a sounds folder and a `db.json` file in it as explained. After that you should be able to start the bot from that folder.

Other than that, I'd need to know what exactly you did and what platform you are on. Maybe a some screenshots of your process could help me to investigate it further.

**Comment by markokajzer** [comment]
`docker compose-up` is working as well for me. Could you add a screenshot of the actual error? How has it `also failed`?

**Comment by nivintw** [comment]
```
cd discord-soundbot
cd config
cp config.example.json config.json
... edit config.json to have my clientId and token...
docker-compose up

... a bunch of normal docker build output ...

Successfully built b9246768f02a
Successfully tagged soundbot:latest
WARNING: Image for service discord-soundbot was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
Creating discord-soundbot_discord-soundbot_1 ... done
Attaching to discord-soundbot_discord-soundbot_1
discord-soundbot_1  | yarn run v1.22.4
discord-soundbot_1  | $ node dist/bin/soundbot.js
discord-soundbot_1  | /app/node_modules/lowdb/adapters/FileSync.js:42
discord-soundbot_1  |           throw e;
discord-soundbot_1  |           ^
discord-soundbot_1  |
discord-soundbot_1  | Error: EISDIR: illegal operation on a directory, read
discord-soundbot_1  |     at Object.readSync (fs.js:564:3)
discord-soundbot_1  |     at tryReadSync (fs.js:349:20)
discord-soundbot_1  |     at readFileSync (fs.js:386:19)
discord-soundbot_1  |     at FileSync.read (/app/node_modules/lowdb/adapters/FileSync.js:35:22)
discord-soundbot_1  |     at LodashWrapper.db.read (/app/node_modules/lowdb/lib/main.js:32:21)
discord-soundbot_1  |     at Object.module.exports [as default] (/app/node_modules/lowdb/lib/main.js:51:13)
discord-soundbot_1  |     at Object.<anonymous> (/app/dist/src/util/db/connection.js:9:35)
discord-soundbot_1  |     at Module._compile (internal/modules/cjs/loader.js:1138:30)
discord-soundbot_1  |     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1158:10)
discord-soundbot_1  |     at Module.load (internal/modules/cjs/loader.js:986:32) {
discord-soundbot_1  |   errno: -21,
discord-soundbot_1  |   syscall: 'read',
discord-soundbot_1  |   code: 'EISDIR'
discord-soundbot_1  | }
discord-soundbot_1  | error Command failed with exit code 1.
discord-soundbot_1  | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
discord-soundbot_discord-soundbot_1 exited with code 1
```

That's the error with clean copy of everything and running `docker-compose up`

Deleting the docker container `discord-soundbot_discord-soundbot_1` that was built from docker compose and deleting the image `soundbot` also so everything is fresh again.

2) Trying the method in the readme pulling the container from dockerhub 
```
... Copying the command in the readme and updating with my clientid and token
output: 

...normal output until...
52ee72c7ffef: Pull complete
be92882e0403: Pull complete
f4ad11e9a8dd: Pull complete
Digest: sha256:3231b260b863b2f491dae182a1c3ac5568204f41f5e36837e43b9a05060c5a2f
Status: Downloaded newer image for markokajzer/discord-soundbot:latest
docker.io/markokajzer/discord-soundbot:latest
yarn run v1.22.4
$ node dist/bin/soundbot.js
/app/node_modules/lowdb/adapters/FileSync.js:42
          throw e;
          ^

Error: EISDIR: illegal operation on a directory, read
    at Object.readSync (fs.js:524:3)
    at tryReadSync (fs.js:349:20)
    at readFileSync (fs.js:386:19)
    at FileSync.read (/app/node_modules/lowdb/adapters/FileSync.js:35:22)
    at LodashWrapper.db.read (/app/node_modules/lowdb/lib/main.js:32:21)
    at Object.module.exports [as default] (/app/node_modules/lowdb/lib/main.js:51:13)
    at Object.<anonymous> (/app/dist/src/util/db/connection.js:9:35)
    at Module._compile (internal/modules/cjs/loader.js:1133:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1153:10)
    at Module.load (internal/modules/cjs/loader.js:977:32) {
  errno: -21,
  syscall: 'read',
  code: 'EISDIR'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Maybe you have something cached locally that's causing it to succeed?

**Comment by nivintw** [comment]
I'm on macOSX 10.15.5, with docker 19.03.8.

**Comment by nivintw** [comment]
I'd love to provide more info, but what i've pasted is all the info that's reported on the command line. With this all being from a clean pull of the repo, i'm not sure what else could be going on.

**Comment by markokajzer** [comment]
I see. I'm on a MacBook as well. I can confirm that `docker-compose up` works.

However, it seems like sometimes db.json is created AS A FOLDER, instead of a FILE. I don't know why that happens unfortunately.

Solution: Delete `db.json` (the folder) and create an empty `db.json` file in the root directory. Rerun `docker run ...` and it should work.

Let me know if that solves the issue

**Comment by markokajzer** [comment]
It seems like this happens, if the `db.json` file does not already exist, so make sure to create it before running `docker run` for the first time, just like described in the installation guide.

**Comment by nivintw** [comment]
I see, i'll give that a try! While I was waiting, I went to the parent directory of where I had cloned the repo, created a completely new empty folder, changed into it, and then ran the docker commands with everything cleaned and it did work! So it's good to know that it works without cloning at all, which I suspected but wasn't sure of. 

I also cleaned up everything and re-cloned, just to see if touching the db.json file before running `docker-compose up` would work. I can confirm, that as long as the config.json file and the db.json file are created then `docker-compose up` from inside the repo clone works! 

Thanks so much for the rapid response; I plan to use this tonight and am grateful for the help!
